projects/packages/scheduled-updates/tests/php/class-wpcom-rest-api-v2-endpoint-update-schedules-test.php

```diff
@@ -796,26 +856,130 @@ public function test_crud_should_sync_only_three_times() {
 
 		$this->assertSame( $base_schedule['timestamp'], $data['timestamp'] );
 		// No sync during read.
-		$this->assertSame( 2, self::$sync_counter );
+		$this->assertSame( 2, self::get_sync_counter() );
 
 		// Delete.
 		$request = new WP_REST_Request( 'DELETE', '/wpcom/v2/update-schedules/' . $schedule_id );
 		$result  = rest_do_request( $request );
 
 		$this->assertTrue( $result->get_data() );
 		// One sync during delete.
-		$this->assertSame( 3, self::$sync_counter );
+		$this->assertSame( 3, self::get_sync_counter() );
+		$this->assertSame( 2, self::$scheduled_counter );
+		$this->assertSame( 3, self::$transients_added );
 	}
 
 	/**
-	 * A callback run when a sync is performed.
+	 * Test additional scheduled event does not reset cron option
+	 */
+	public function test_another_scheduled_event_does_not_reset() {
+		$plugins = array(
+			'custom-plugin/custom-plugin.php',
+			'gutenberg/gutenberg.php',
+		);
+		$request = new WP_REST_Request( 'POST', '/wpcom/v2/update-schedules' );
+		$request->set_body_params(
+			array(
+				'plugins'  => $plugins,
+				'schedule' => $this->get_schedule(),
+			)
+		);
+
+		// Successful request.
+		wp_set_current_user( $this->admin_id );
+
+		$result      = rest_do_request( $request );
+		$schedule_id = Scheduled_Updates::generate_schedule_id( $plugins );
+
+		$this->assertSame( 200, $result->get_status() );
+		$this->assertSame( $schedule_id, $result->get_data() );
+
+		$this->assertArrayHasKey( $schedule_id, wp_get_scheduled_events( Scheduled_Updates::PLUGIN_CRON_HOOK ) );
+		$this->assertSame( 1, self::get_sync_counter() );
+		$this->assertSame( 1, self::$scheduled_counter );
+
+		$plugins[] = 'wp-test-plugin/wp-test-plugin.php';
+		$request   = new WP_REST_Request( 'PUT', '/wpcom/v2/update-schedules/' . $schedule_id );
+		$request->set_body_params(
+			array(
+				'plugins'  => $plugins,
+				'schedule' => $this->get_schedule(),
+			)
+		);
+
+		$result      = rest_do_request( $request );
+		$schedule_id = Scheduled_Updates::generate_schedule_id( $plugins );
+
+		$this->assertSame( 200, $result->get_status() );
+		$this->assertArrayHasKey( $schedule_id, wp_get_scheduled_events( Scheduled_Updates::PLUGIN_CRON_HOOK ) );
+		$this->assertSame( 2, self::get_sync_counter() );
+		$this->assertSame( 2, self::$scheduled_counter );
+		$this->assertSame( 2, self::$transients_added );
+
+		// Insert a full cron sync event.
+		wp_schedule_event( time(), 'jetpack_sync_interval', 'jetpack_sync_full_cron' );
+
+		$request = new WP_REST_Request( 'GET', '/wpcom/v2/update-schedules' );
+		$result  = rest_do_request( $request );
+
+		$this->assertSame( 200, $result->get_status() );
```

Comment:
The test is to check if the cron gets reset after another schedule event, but I not sure what's the purpose of having a full cron sync event here to test for, am I missing something?

Reply:
With our latest changes, this unit test can be removed.